### PR TITLE
Features/two-phase generation

### PR DESCRIPTION
**Pull request by beorosz**
Make possible to serialize/deserialize SigSpecDocument

### CODE REVIEW

File: src/SigSpec.Core/SigSpecDocument.cs [hunk 1]
```diff
@@ -2,13 +2,11 @@
 using Newtonsoft.Json.Converters;
 using Newtonsoft.Json.Serialization;
 using NJsonSchema;
-using NJsonSchema.Converters;
 using System;
 using System.Collections.Generic;
 
 namespace SigSpec.Core
 {
-    [JsonConverter(typeof(JsonReferenceConverter))]
```
Comment (RicoSuter) [inline src/SigSpec.Core/SigSpecDocument.cs]: Why do you remove this? I think it's needed for correct $ref handling

Comment (beorosz) [reply src/SigSpec.Core/SigSpecDocument.cs]: I wasn't able to properly serialize and then deserialize SigSpecDocument. Deserialization always threw StackOverflowException. It looked has an infinite loop while trying to deserialize the document (commit aeeaf90 depicts the problem) . Removing the attribute solved the problem and I was able to deserialize and generate the TS client.

Comment (beorosz) [reply src/SigSpec.Core/SigSpecDocument.cs]: If you think my solution does not work, please indicate and I revoke my PR.

**Comment by beorosz** [comment]
Closing the PR as it is not accepted.

**Pull request closed without merging**